{
	"ID": "202305-02",
	"Title": "Python, PyPy3: Multiple Vulnerabilities",
	"Synopsis": "Multiple vulnerabilities have been found in Python and PyPy, the worst of which could result in arbitrary code execution.",
	"Product": {
		"Text": "pypy3,python",
		"Type": "ebuild"
	},
	"Announced": "2023-05-03",
	"Revised": {
		"Text": "2023-05-03",
		"Count": "1"
	},
	"Bug": [
		"880629",
		"878385",
		"877851",
		"876815",
		"864747",
		"838250",
		"835443",
		"834533",
		"787260",
		"811165",
		"793833"
	],
	"Access": "local and remote",
	"Affected": {
		"Package": [
			{
				"Name": "dev-lang/python",
				"Auto": "yes",
				"Arch": "*",
				"Unaffected": [
					{
						"Text": "3.8.15_p3",
						"Range": "ge",
						"Slot": "3.8"
					},
					{
						"Text": "3.9.15_p3",
						"Range": "ge",
						"Slot": "3.9"
					},
					{
						"Text": "3.10.8_p3",
						"Range": "ge",
						"Slot": "3.10"
					},
					{
						"Text": "3.11.0_p2",
						"Range": "ge",
						"Slot": "3.11"
					},
					{
						"Text": "3.12.0_alpha1_p2",
						"Range": "ge",
						"Slot": "12"
					}
				],
				"Vulnerable": [
					{
						"Text": "3.8.15_p3",
						"Range": "lt",
						"Slot": "3.8"
					},
					{
						"Text": "3.9.15_p3",
						"Range": "lt",
						"Slot": "3.9"
					},
					{
						"Text": "3.10.8_p3",
						"Range": "lt",
						"Slot": "3.10"
					},
					{
						"Text": "3.11.0_p2",
						"Range": "lt",
						"Slot": "3.11"
					},
					{
						"Text": "3.12.0_alpha1_p2",
						"Range": "lt",
						"Slot": "12"
					}
				]
			},
			{
				"Name": "dev-python/pypy3",
				"Auto": "yes",
				"Arch": "*",
				"Unaffected": [
					{
						"Text": "7.3.9_p9",
						"Range": "ge",
						"Slot": ""
					}
				],
				"Vulnerable": [
					{
						"Text": "7.3.9_p9",
						"Range": "lt",
						"Slot": ""
					}
				]
			}
		],
		"Service": {
			"Text": "",
			"Type": "",
			"Fixed": ""
		}
	},
	"Background": {
		"Text": "\n        <p>Python is an interpreted, interactive, object-oriented, cross-platform programming language.</p>\n    "
	},
	"Description": {
		"Text": "\n        <p>Multiple vulnerabilities have been discovered in Python and PyPy3. Please review the CVE identifiers referenced below for details.</p>\n    "
	},
	"Impact": {
		"Type": "high",
		"Text": "\n        <p>Please review the referenced CVE identifiers for details.</p>\n    "
	},
	"Workaround": {
		"Text": "\n        <p>There is no known workaround at this time.</p>\n    "
	},
	"Resolution": {
		"Text": "\n        \n        \n        \n        \n        \n        \n        \n        \n        \n        \n        \n        \n        \n        \n        \n        \n        \n        \n        \n        \n        \n        \n        \n    ",
		"P": [
			{
				"Text": "All Python 3.8 users should upgrade to the latest version:",
				"URI": {
					"Text": "",
					"Link": ""
				},
				"Br": null,
				"I": null,
				"Code": "",
				"P": ""
			},
			{
				"Text": "All Python 3.9 users should upgrade to the latest version:",
				"URI": {
					"Text": "",
					"Link": ""
				},
				"Br": null,
				"I": null,
				"Code": "",
				"P": ""
			},
			{
				"Text": "All Python 3.10 users should upgrade to the latest version:",
				"URI": {
					"Text": "",
					"Link": ""
				},
				"Br": null,
				"I": null,
				"Code": "",
				"P": ""
			},
			{
				"Text": "All Python 3.11 users should upgrade to the latest version:",
				"URI": {
					"Text": "",
					"Link": ""
				},
				"Br": null,
				"I": null,
				"Code": "",
				"P": ""
			},
			{
				"Text": "All Python 3.12 users should upgrade to the latest version:",
				"URI": {
					"Text": "",
					"Link": ""
				},
				"Br": null,
				"I": null,
				"Code": "",
				"P": ""
			},
			{
				"Text": "All PyPy3 users should upgrade to the latest version:",
				"URI": {
					"Text": "",
					"Link": ""
				},
				"Br": null,
				"I": null,
				"Code": "",
				"P": ""
			}
		],
		"Code": [
			"\n          # emerge --sync\n          # emerge --ask --oneshot --verbose \">=dev-lang/python-3.8.15_p3:3.8\"\n        ",
			"\n          # emerge --sync\n          # emerge --ask --oneshot --verbose \">=dev-lang/python-3.9.15_p3:3.9\"\n        ",
			"\n          # emerge --sync\n          # emerge --ask --oneshot --verbose \">=dev-lang/python-3.10.8_p3:3.10\"\n        ",
			"\n          # emerge --sync\n          # emerge --ask --oneshot --verbose \">=dev-lang/python-3.11.0_p2:3.11\"\n        ",
			"\n          # emerge --sync\n          # emerge --ask --oneshot --verbose \">=dev-lang/python-3.12.0_alpha1_p2\"\n        ",
			"\n          # emerge --sync\n          # emerge --ask --oneshot --verbose \">=dev-python/pypy3-7.3.9_p9\"\n        "
		]
	},
	"References": {
		"URI": [
			{
				"Text": "CVE-2015-20107",
				"Link": "https://nvd.nist.gov/vuln/detail/CVE-2015-20107"
			},
			{
				"Text": "CVE-2021-3654",
				"Link": "https://nvd.nist.gov/vuln/detail/CVE-2021-3654"
			},
			{
				"Text": "CVE-2021-28363",
				"Link": "https://nvd.nist.gov/vuln/detail/CVE-2021-28363"
			},
			{
				"Text": "CVE-2021-28861",
				"Link": "https://nvd.nist.gov/vuln/detail/CVE-2021-28861"
			},
			{
				"Text": "CVE-2021-29921",
				"Link": "https://nvd.nist.gov/vuln/detail/CVE-2021-29921"
			},
			{
				"Text": "CVE-2022-0391",
				"Link": "https://nvd.nist.gov/vuln/detail/CVE-2022-0391"
			},
			{
				"Text": "CVE-2022-37454",
				"Link": "https://nvd.nist.gov/vuln/detail/CVE-2022-37454"
			},
			{
				"Text": "CVE-2022-42919",
				"Link": "https://nvd.nist.gov/vuln/detail/CVE-2022-42919"
			},
			{
				"Text": "CVE-2022-45061",
				"Link": "https://nvd.nist.gov/vuln/detail/CVE-2022-45061"
			}
		]
	},
	"Metadata": [
		{
			"Text": "ajak",
			"Tag": "requester",
			"Timestamp": "2023-05-03T09:12:43.325618Z"
		},
		{
			"Text": "sam",
			"Tag": "submitter",
			"Timestamp": "2023-05-03T09:12:43.330732Z"
		}
	]
}
